iroh-net/src/hp/netcheck.rs

```diff
@@ -693,14 +693,16 @@ impl ReportState {
         }
 
         // abort the rest of the probes
-        debug!("aborting {} probes, already done", probes.len());
+        if !probes.is_empty() {
+            debug!("aborting {} probes, already done", probes.len());
+        }
         drop(probes);
 
         if let Some(hair_pin) = self.wait_hair_check().await {
             self.report.write().await.hair_pinning = Some(hair_pin);
         }
 
-        if !skip_external_network && port_mapper.is_some() {
+        if do_port_map {
```

Comment:
so we start portmapping anyway as long as we're allows to use an external network, but don't always bother waiting for it?  what's the logic for that?

Reply:
I'm not sure what you are referring to. The logic is akin to what was in here already. In reality we could just do `port_mapping.inner.is_some()`

Reply:
yeah, i know the logic didn't change.  i was only trying to understand myself why it was this way and given you modified it maybe you knew...

Reply:
ah I see. My understanding is that we always finish it actually. At least that's my understanding from tailscale's code. Code as is pretends to do it with the line just below this one without actually polling the future. But in theory, yes, we should finish the probe